tailnet/conn_test.go

```diff
@@ -412,9 +413,78 @@ parentLoop:
 	require.True(t, client2.AwaitReachable(awaitReachableCtx4, ip))
 }
 
+func TestConn_BlockEndpoints(t *testing.T) {
+	t.Parallel()
+	logger := slogtest.Make(t, nil).Leveled(slog.LevelDebug)
+
+	derpMap, _ := tailnettest.RunDERPAndSTUN(t)
+
+	// Setup conn 1.
+	ip1 := tailnet.IP()
+	conn1, err := tailnet.NewConn(&tailnet.Options{
+		Addresses:      []netip.Prefix{netip.PrefixFrom(ip1, 128)},
+		Logger:         logger.Named("w1"),
+		DERPMap:        derpMap,
+		BlockEndpoints: true,
+	})
+	require.NoError(t, err)
+	defer func() {
+		err := conn1.Close()
+		assert.NoError(t, err)
+	}()
+
+	// Setup conn 2.
+	ip2 := tailnet.IP()
+	conn2, err := tailnet.NewConn(&tailnet.Options{
+		Addresses:      []netip.Prefix{netip.PrefixFrom(ip2, 128)},
+		Logger:         logger.Named("w2"),
+		DERPMap:        derpMap,
+		BlockEndpoints: true,
+	})
+	require.NoError(t, err)
+	defer func() {
+		err := conn2.Close()
+		assert.NoError(t, err)
+	}()
+
+	// Connect them together and wait for them to be reachable.
+	nodes1 := stitch(t, conn2, conn1)
+	nodes2 := stitch(t, conn1, conn2)
+	awaitReachableCtx, awaitReachableCancel := context.WithTimeout(context.Background(), testutil.WaitShort)
+	defer awaitReachableCancel()
+	require.True(t, conn1.AwaitReachable(awaitReachableCtx, ip2))
+
+	// Watch for 10 seconds to ensure that neither peer has any endpoints.
+	// There's no good way to force endpoint updates, so we just wait.
+parentLoop:
+	for {
+		select {
+		case node := <-nodes1:
+			require.Empty(t, node.Endpoints, "conn1 got endpoints")
+		case node := <-nodes2:
+			require.Empty(t, node.Endpoints, "conn2 got endpoints")
+		case <-time.After(testutil.WaitShort):
```

Comment:
timer gets reset each time we go around the loop, so we could wait for much longer than 10 sec.

But, I don't think we add much value by waiting for 10s.  There are unit tests for `tailnet.Conn` blocking endpoints on node updates.  

What wasn't covered prior to this is endpoints discovered via disco.  The status tests you have below cover that.

Reply:
Oops, yeah I guess I can just remove this section then. I do still think we should wait because we can't force it to gather local endpoints, unless you know a way to test this reliably?

Reply:
we have tests that demonstrate that `tailnet.Conn` won't send endpoint in a Node update callback when blockendpoints is set.

E.g. `TestNodeUpdater_setBlockEndpoints_different`

Where we got the endpoints from (local, STUN, etc) is immaterial.

Reply:
I removed the node endpoint check but I still think the delay is required unless you know of another way to force disco endpoints to be gathered and sent immediately